inline_verifier.go

```diff
@@ -326,14 +326,24 @@ func (v *InlineVerifier) CheckFingerprintInline(tx *sql.Tx, targetSchema, target
 
 			compressedData, ok = rowData[idx].([]byte)
 			if !ok {
-				return nil, fmt.Errorf("cannot convert column %v to []byte", col.Name)
+				return fmt.Errorf("cannot convert column %v to []byte", col.Name)
 			}
 
 			sourceDecompressedData[paginationKey][col.Name], err = v.decompressData(table, col.Name, compressedData)
 		}
 	}
 
-	return v.compareHashesAndData(sourceFingerprints, targetFingerprints, sourceDecompressedData, targetDecompressedData), nil
+	mismatches := v.compareHashesAndData(sourceFingerprints, targetFingerprints, sourceDecompressedData, targetDecompressedData)
+
+	for _, mismatchedPk := range mismatches {
+		v.reverifyStore.Add(table, mismatchedPk)
+	}
+
+	if len(mismatches) > 0 {
+		v.logger.WithField("mismatches", mismatches).Info("inline verification during data copy noticed mismatched pk, which is okay (issue 149)")
+	}
+
+	return nil
```

Comment:
This should end the main code changes. I added a lot of tests and had to change a lot of them.